src/index.js

```diff
@@ -176,7 +188,14 @@ export default ({
   let submit = Command(() => {
     if (_.isEmpty(form.validate())) {
       form.submit.state.error = null
-      return configSubmit(form.getSnapshot(), form)
+      try {
+        // Handle both sync and sync configSubmit
+        return Promise.resolve(configSubmit(form.getSnapshot(), form)).catch(
+          err => handleSubmitErr(state, err)
+        )
```

Comment:
Looks like I was wrong. `Command` does support async functions, so you don't have to deal with `Promise.resolve` and `catch`. Also, you can use

```
        if (err instanceof ValidationError) {
          state.errors = {
            '': [err.message],
            ...err.cause,
          }
        }
```

to use the error message as the error message for the entire form.